packages/metro-react-native-babel-preset/src/configs/main.js

```diff
@@ -115,20 +102,6 @@ const getPreset = (src, options) => {
   if (isHermes && (isNull || src.indexOf('async') !== -1)) {
     extraPlugins.push([require('@babel/plugin-transform-async-to-generator')]);
   }
-  if (!isHermes && (isNull || src.indexOf('**') !== -1)) {
-    extraPlugins.push([
-      require('@babel/plugin-transform-exponentiation-operator'),
-    ]);
-  }
-  if (!isHermes && (isNull || src.indexOf('Object.assign')) !== -1) {
-    extraPlugins.push([require('@babel/plugin-transform-object-assign')]);
-  }
-  if (!isHermes && hasForOf) {
-    extraPlugins.push([
-      require('@babel/plugin-transform-for-of'),
```

Comment:
We have some internal tests that are now failing because the `for-of` transform is a prerequisite for `regenerator` (https://github.com/facebook/regenerator/issues/229). Have you looked into whether we can remove `regenerator` altogether for JSC? (We are already skipping it in `hermes-stable` and `hermes-canary`)

Reply:
I believe we can - we should be able to use `async-to-generator` at least. I will try it out.

Reply:
@motiz88 I was able to replace regenerator with `async-to-generator` and didn't have any issues with async code in our app. Code and tests updated. Thanks!